modules/integration/src/test/scala/scala/cli/integration/RunScriptTestDefinitions.scala

```diff
@@ -326,4 +326,18 @@ trait RunScriptTestDefinitions { _: RunTestDefinitions =>
         expect(p.out.trim() == "List(1, 2, 3, -v)")
       }
     }
+
+  test("script file with shebang line and no extension run with scala-cli shebang") {
```

Comment:
we might want an extra sanity check verifying that an input without the shebang header can't be run if it has no supported extension.
so another test beyond this one, maybe?